hardbyte/python-can
# 1210 Add compressed file methods

This closes #1210 by adding `LogReader.decompress` / `Logger.compress` methods.

Changes:
- ~~add `io/gz.py`~~
- ~~add `GzipReader` class~~
- ~~add `GzipReader` to `LogReader` message readers~~
- edit `LogReader` to check for last suffix only
- edit `LogReader` docstring 
- add `decompress` method to `LogReader`
- add unittest for compressed player
- edit `Logger` to check for last suffix only
- edit `Logger` docstring 
- add `compress` method to `Logger`
- add unittest for compressed logger
- add multiple dot suffix to `test_extension_matching`
- remove `GzipASCReader` / `GzipASCWriter` and subsequent unittests
